python/mxnet/ndarray.py

```diff
@@ -560,24 +559,6 @@ def wait_to_read(self):
         check_call(_LIB.MXNDArrayWaitToRead(self.handle))
 
 
-    @property
-    def shape(self):
```

Comment:
any such examples? NDArray interface has been fairly stable.
The problem with SparseNDArray not being a NDArray is all the places where you use isinstance(x, NDArray) need to be changed to isinstance(x, NDArrayBase)